### PR TITLE
Standardise on hyphenated UUIDS

### LINKED ISSUE #995: inconsistency in UUID formatting
in #959, @titanous brought up the point that Flynn strips dashes out of UUIDs at the presentation layer:

> In the controller we strip the dashes from UUIDs before they hit the presentation layer. I can't remember the rationale for it, and I'm open to reversing that decision, but that should happen on another ticket.

### LINKED ISSUE #1796: controller route id inconsistency
Route events coming through the controller remove the hyphens from the ids, but all other endpoints return the hyphenated route id as returned by the router.

#1784 fixes this by returning hyphenated ids everywhere.

### PR DESCRIPTION
**Pull request by josephglanville**
Closes #995
Closes #1796

### CODE REVIEW

File: router/http_test.go [hunk 1]
```diff
@@ -279,7 +279,9 @@ func httpHeaderTestHandler(c *C, ip, port string) http.Handler {
 		c.Assert(req.Header["X-Forwarded-Proto"][0], Equals, "http")
 		c.Assert(len(req.Header["X-Request-Start"][0]), Equals, 13)
 		c.Assert(req.Header["X-Forwarded-For"][0], Equals, ip)
-		c.Assert(len(req.Header["X-Request-Id"][0]), Equals, 32)
+		// size of UUID + the 4 hyphens
+		// maybe better to actually parse the UUID back to bytes here?
```
Comment (josephglanville) [inline router/http_test.go]: I kind of feel like we should parse the UUID here for completeness rather than just counting bytes, thoughts?

Comment (titanous) [inline router/http_test.go]: You could use `fmt.Scan*` to avoid writing a parser

Comment (josephglanville) [inline router/http_test.go]: Decided to just use the regex matcher here, simple and ensure it's valid.

**Comment by titanous** [comment]
The commit message needs an `all: ` prefix.

**Comment by josephglanville** [comment]
ping @titanous

**Comment by titanous** [comment]
LGTM. Thanks!

**Pull request merged**